packages/engine/src/scene/functions/ErrorFunctions.ts

```diff
@@ -45,7 +45,7 @@ export const addError = <C extends Component>(
   message?: string
 ) => {
   console.error('[addError]:', entity, Component.name, error, message)
-  if (!hasComponent(entity, ErrorComponent)) addComponent(entity, ErrorComponent)
+  if (!hasComponent(entity, ErrorComponent)) setComponent(entity, ErrorComponent)
```

Comment:
We also need logic to remove the error component. At the moment the last error set remains even when the asset loader does its thing successfully.